nixos/modules/programs/wayland/river.nix

```diff
@@ -1,26 +1,29 @@
-{
-  config,
-  pkgs,
-  lib,
-  ...
-}:
-with lib; let
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
   cfg = config.programs.river;
-in {
+in
+{
   options.programs.river = {
     enable = mkEnableOption (lib.mdDoc "river, a dynamic tiling Wayland compositor");
 
     package = mkOption {
       type = with types; nullOr package;
```

Comment:
Why is this option nullOr? If we gate things behind the enable option we can remove this.

Reply:
This comes from the Sway module, which this module is based on. The description says we should set it to null if we use the HM module (nix-community/home-manager#3977) to install River.

Reply:
That is a bit weird and feels hacky. Can we avoid this?

Reply:
The `package` option in the Sway module was introduced in #213482 and copied as is in the River module, so see the rationale in this PR (which I'm not sure to understand).

In the same vein, I would also be in favour of removing the `extraPackages` option in both the Sway and River modules, as they are just a duplicate of `environment.systemPackages`.

I didn't look much further though, because I didn't want any breaking change in this PR.